### PR TITLE
Use getLocationOfThisType instead of embedGenericHandle

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
getLocationOfThisType does way less work and I believe it gives the same result (the part we care about).

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
@sandreenko @pgavlin PTAL

(I was messing with the RyuJIT codebase when I rediscovered this EE interface API by accident. I believe @sandreenko only used `embedGenericHandle` because my lousy prototype used it.)

File: src/jit/compiler.h [hunk 1]
```diff
@@ -2912,7 +2912,7 @@ class Compiler
                                unsigned                flags,
                                void*                   compileTimeHandle);
 
-    GenTreePtr getRuntimeContextTree(CORINFO_LOOKUP* pLookup);
+    GenTreePtr getRuntimeContextTree(CORINFO_RUNTIME_LOOKUP_KIND pLookup);
```
Comment (jkotas) [inline src/jit/compiler.h]: Nit: pLookup -> kind

File: src/jit/importer.cpp [hunk 2]
```diff
@@ -1860,18 +1860,17 @@ GenTreePtr Compiler::impMethodPointer(CORINFO_RESOLVED_TOKEN* pResolvedToken, CO
...
 // Return Value:
 //    Return GenTree pointer to generic shared context.
 //
 // Notes:
 //    Reports about generic context using.
 
-GenTreePtr Compiler::getRuntimeContextTree(CORINFO_LOOKUP* pLookup)
+GenTreePtr Compiler::getRuntimeContextTree(CORINFO_RUNTIME_LOOKUP_KIND kind)
 {
     GenTreePtr                  ctxTree = nullptr;
```
Comment (jkotas) [inline src/jit/importer.cpp]: The code formatting enforcement insist on having just a single space between `GenTreePtr` and `ctxTree` after your change.

**Comment by MichalStrehovsky** [comment]
@dotnet/jit-contrib Can someone have a look please?

**Comment by CarolEidt** [comment]
LGTM

**Pull request merged**